packages/js/product-editor/src/bindings-sources/entity/product/index.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import { useEntityProp } from '@wordpress/core-data';
+import { __ } from '@wordpress/i18n';
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	BindingSourceHandlerProps,
+	BindingUseSourceProps,
+	BlockProps,
+} from '../../../bindings/types';
+import { WooEntitySourceArgs } from './types';
+
+/**
+ * React custom hook to bind a source to a block.
+ *
+ * @param {BlockProps}          blockProps - The block props.
+ * @param {WooEntitySourceArgs} sourceArgs - The source args.
+ * @return {BindingUseSourceProps} The source value and setter.
+ */
+const useSource = (
+	blockProps: BlockProps,
+	sourceArgs: WooEntitySourceArgs
```

Comment:
`WooEntitySourceArgs` has `kind` and `name` props, but these aren't used in here. So, it makes me think that we should use a different type that is more focused, such as `WooCommerceEntityProductSourceArgs` (side note: I think we should use `WooCommerce` instead `Woo`, because `Woo` refers to the company and `WooCommerce` refers to the open-source product).

Reply:
Thanks. Nice feedback. Addressed  here https://github.com/woocommerce/woocommerce/pull/45616/commits/1201cc1a3557c61fb7d361d15985282dbef819b3